x-pack/plugins/ingest_manager/common/services/is_valid_namespace.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// Namespace string eventually becomes part of an index name. This method partially implements index name rules from
+// https://github.com/elastic/elasticsearch/blob/master/docs/reference/indices/create-index.asciidoc
+export function isValidNamespace(namespace: string) {
+  return (
+    typeof namespace === 'string' &&
+    // Lowercase only
+    namespace === namespace.toLowerCase() &&
```

Comment:
just a few cases missing which agent validates
- namespace is allowed to contain `.` but is not allowed to be equal to `.` or `..`
- namespace cannot be longer than 255 chars
- namespace cannot start with `-`, `_` or `+`

Reply:
@michalpristas I think the validation is incorrect, 


```
Cannot be longer than 255 bytes (note it is bytes, so multi-byte characters will count towards the 255 limit faster)
```

This mean, type + dataset + namespace <= 255 bytes.

Reply:
what we do in agent is that we put 255 constraint on dataset, namespace and a product of concatenation, so eventaully we will fail when index > 255.
putting some not communicated constraint felt weird (such as 255/3 or some other math) so i used 255 on each part and result as well. 
as namespace is created before configuration and can lead to different indexes based on dataset and type it would make sense to agree on some constraint which can be applied at the time of specifying namespace.

Reply:
cc @jen-huang

Reply:
I followed up in https://github.com/elastic/kibana/issues/75846#issuecomment-680092618.